src/VisualStudio/Core/Def/Implementation/Preview/FileChange.cs

```diff
@@ -79,13 +80,18 @@ private ChangeList ComputeChildren(TextDocument left, TextDocument right, Cancel
 
             diffService = diffService ?? diffSelector.DefaultTextDifferencingService;
 
-            var diff = ComputeDiffSpans(oldText, newText, diffService, cancellationToken);
-            if (!diff.Differences.Any())
+            var diff = ComputeDiffSpans(diffService, left, right, cancellationToken);
+            if (diff.Differences.Count == 0)
             {
-                // There are no changes. Just show the whole document.
-                return GetEntireDocumentAsSpanChange(left);
+                // There are no changes.
+                return ChangeList.Empty;
```

Comment:
@mavasani 

well, I didn't want to spend too much time refactoring this code. the way code is current structured. it does all computation on constructor. not factory, not outside of constructor. so this was only way to communicate that there is no change without many refactoring..

anyway, outer, I check the file change has any change in it. if it doesn't I don't put them in the list.

this is final form you see when there is no change.

![image](https://cloud.githubusercontent.com/assets/1333179/7261774/2376a3e8-e829-11e4-9162-213fe99ee1ee.png)

...

by the way, this code is not specific to code fix or code action. this is for any consumer such as rename, extract method and all.